Node.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BST
+{
+    class Node
+    {
+        public int key;

```

Comment:
This does not follow encapsulation patterns of OOP. Encapsulation would restrict access to vital attributes. And for example, here, I'm not sure if enabling other classes/developers to change value of the leaf freely would not end up in unexpected behavior of class. Same goes for 'left', 'right' attributes of this class.